Content.Server/Traits/TraitSystem.Functions.cs

```diff
@@ -698,3 +701,58 @@ public override void OnPlayerSpawn(EntityUid uid,
         tagSystem.AddTags(uid, Tags);
     }
 }
+
+// <summary>
+//      Replaces a body part with a cybernetic. This is only for limbs such as arms and legs, don't use this for organs(old or new).
+// </summary>
+[UsedImplicitly]
+public sealed partial class TraitCyberneticLimbReplacement : TraitFunction
+{
+    [DataField, AlwaysPushInheritance]
+    public BodyPartType RemoveBodyPart { get; private set; } = BodyPartType.Arm;
+
+    [DataField, AlwaysPushInheritance]
+    public BodyPartSymmetry PartSymmetry { get; private set; } = BodyPartSymmetry.Left;
+
+    [DataField, AlwaysPushInheritance]
+    public EntProtoId? ProtoId { get; private set; }
+
+    [DataField, AlwaysPushInheritance]
+    public string SlotId { get; private set; } = "right arm";
+
+    public override void OnPlayerSpawn(EntityUid uid,
+        IComponentFactory factory,
+        IEntityManager entityManager,
+        ISerializationManager serializationManager)
+    {
+        var bodySystem = entityManager.System<BodySystem>();
+        var transformSystem = entityManager.System<SharedTransformSystem>();
+
+        if (!entityManager.TryGetComponent(uid, out BodyComponent? body)
+            || !entityManager.TryGetComponent(uid, out TransformComponent? xform)
+            || ProtoId is null)
+            return;
+
+        var root = bodySystem.GetRootPartOrNull(uid, body);
+        if (root is null)
+            return;
+
+        var parts = bodySystem.GetBodyChildrenOfType(uid, RemoveBodyPart, body);
+        foreach (var part in parts)
+        {
+            var partComp = part.Component;
+            if (partComp.Symmetry != PartSymmetry)
+                continue;
+
+            foreach (var child in bodySystem.GetBodyPartChildren(part.Id, part.Component))
+                entityManager.QueueDeleteEntity(child.Id);
+
+            transformSystem.AttachToGridOrMap(part.Id);
+            entityManager.QueueDeleteEntity(part.Id);
+
+            var newLimb = entityManager.SpawnAtPosition(ProtoId, xform.Coordinates);
```

Comment:
```suggestion
            var newLimb = entityManager.SpawnAtPosition(ProtoId, transform.Coordinates);
```
This doesn't seem like a clear abbreviation--code is _read_ more often than it is written, so abbreviating like this might not be helpful for other contributed seeking to abuse this system later